lib/Autocomplete.js

```diff
@@ -7,20 +7,106 @@ let _debugStates = []
 let Autocomplete = React.createClass({
 
   propTypes: {
+    /**
+     * The items to display in the dropdown menu
+     */
+    items: React.PropTypes.array.isRequired,
+    /**
+     * The value to display in the input field
+     */
     value: React.PropTypes.any,
+    /**
+     * Arguments: `event: Event, value: String`
+     *
+     * Invoked every time the user changes the input's value.
+     */
     onChange: React.PropTypes.func,
+    /**
+     * Arguments: `value: String, item: Any`
+     *
+     * Invoked when the user selects an item from the dropdown menu.
+     */
     onSelect: React.PropTypes.func,
+    /**
+     * Arguments: `item: Any, value: String`
+     *
+     * Invoked for each entry in `items` and its return value is used to
+     * determine whether or not it should be displayed in the dropdown menu.
+     * By default all items are always rendered.
```

Comment:
I take it this is to give more control to the passed items on if they display in the list or not? (e.g. when the input value is updated by the user?)